mpd/mpd.go

```diff
@@ -126,6 +125,50 @@ type CommonAttributesAndElements struct {
 	InbandEventStream         *DescriptorType       `xml:"inbandEventStream,attr"`
 }
 
+type contentProtections []ContentProtectioner
+
+func (as *contentProtections) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	var scheme string
+	for _, a := range start.Attr {
+		if a.Name.Local == "schemeIdUri" {
+			scheme = a.Value
+			break
+		}
+	}
+	var target ContentProtectioner
+	switch scheme {
+	case CONTENT_PROTECTION_ROOT_SCHEME_ID_URI:
+		target = &CENCContentProtection{}
+	case CONTENT_PROTECTION_PLAYREADY_SCHEME_ID:
+		target = &PlayreadyContentProtection{}
+	case CONTENT_PROTECTION_WIDEVINE_SCHEME_ID:
+		target = &WidevineContentProtection{}
+	default:
+		target = &ContentProtection{}
+	}
+	if err := d.DecodeElement(target, &start); err != nil {
+		return err
+	}
+	*as = append(*as, target)
+	return nil
+}
+
+// wrappedAdaptationSet provides the default xml unmarshal
+// to take care of the majority of our unmarshalling
+type wrappedAdaptationSet AdaptationSet
+
+// dtoAdaptationSet parses the items out of AdaptationSet
+// that give us trouble:
+// * slices of pointers
+// * Content Protection interface
+type dtoAdaptationSet struct {
+	wrappedAdaptationSet
+	Roles              []Role             `xml:"Role,omitempty"`
```

Comment:
I don't think slices of pointers are causing us any troubles, just the content protections. You should be able to remove `Roles` `Representations` and `AccesibilityElems` from here and let the default unmarshaling of `wrappedAdaptationSet` take care of them

Reply:
oh weird good catch. I swear they were giving me troubles earlier